integration/disk_cleanup_helpers.go

```diff
@@ -151,11 +151,47 @@ func waitUntilDataCleanedUpExtended(
 	return errDataCleanupTimedOut
 }
 
-// nolint: deadcode
-func waitUntilFilesetsCleanedUp(filePathPrefix string, namespace ts.ID, extraShard uint32, waitTimeout time.Duration) error {
+// nolint: deadcode, unused
+func waitUntilNamespacesCleanedUp(testSetup *testSetup, filePathPrefix string, namespace ts.ID, waitTimeout time.Duration) error {
+
+	// The idea would be to have a channel that has notifs (ie, --> func if val 1, 2, or 2)
+	// and pass that to a function that does a function based on that state
+	// this would involve locks on the channel to check the functions aren't called
+	// more than once
+
+	c := make(chan error)
+	go func() {
+		c <- waitUntilNamespacesHaveReset(testSetup)
+	}()
 	dataCleanedUp := func() bool {
-		shardDir := fs.ShardDirPath(filePathPrefix, namespace, extraShard)
-		return !fs.FileExists(shardDir)
+		namespaceDir := fs.NamespaceDirPath(filePathPrefix, namespace)
+		return !fs.FileExists(namespaceDir)
+	}
+
+	if waitUntil(dataCleanedUp, waitTimeout) {
+		return nil
+	}
+	return errDataCleanupTimedOut
+}
+
+// nolint: deadcode, unused
+func waitUntilNamespacesHaveReset(testSetup *testSetup) error {
```

Comment:
Seems `stopServer` is not called here so `waitUntilServerIsDown()` will never finish I believe?